engine/src/flutter/shell/platform/windows/host_window_dialog.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_HOST_WINDOW_DIALOG_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_HOST_WINDOW_DIALOG_H_
+
+#include "host_window.h"
+
+namespace flutter {
+class HostWindowDialog : public HostWindow {
+ public:
+  // Creates a dialog window.
+  //
+  // If |owner_window| is not null, the dialog will be modal to the owner.
```

Comment:
TODO: What does it mean to be modal to the owner? I assume you are destroyed if your owner is destroyed. Anything else?